### ISSUE TITLE
Update wiki from svn via hook

### ISSUE THREAD

**Issue by jbs1** [OpenMath/OM3#85]
_migrated from Trac, where originally posted by **clange** on 13-Sep-2008 10:44pm_

When a change to a CD is committed to the svn, the wiki should be notified by some ping backdoor that I will provide.

**Paul**, once I tell you the URL for that, could you set up a post-commit hook that watches `OpenMath3/cd/MathML/*.{ocd,ntn}` for changes?

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **kohlhase** on 14-Sep-2008 5:57am_

This is a very important thing to get the update of the wiki going. There will always be people who edit the CDs via the svn. A short term alternative would be to run a cron-like job inside SWiM that updates every 5 minutes, or even beter updates a file whenever an edit view is requested. Probably both.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **clange** on 14-Sep-2008 8:14pm_

An update _is'' already done whenever an edit view is requested, as outlined on the [FAQ](http://wiki.openmath.org/?title=FAQ) page.  The only thing that's missing is the update for those who ''view_ pages in the wiki, whenever a page is changed in the svn but not edited in the wiki.

My plan is as follows:
1. provide the API with the "ping" backdoor
2. regularly run this for all pages by a normal cron job on wiki.openmath.org
3. disable that cron job once the svn hook is implemented

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **clange** on 24-Sep-2008 3:55am_

This bug is now obsolete.  The reading view now always checks if there is a newer version in the repository, and if so, the current page is updated.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **clange** on 24-Sep-2008 3:16pm_

This problem is not yet completely solved, as the Subversion client in SWiM currently tries to log into the repository using a "wiki account -> svn account" mapping for any action.  It is not yet capable of falling back to anonymous access (or vice versa, i.e. trying anonymous first, and then connecting as some user).  For you that doesn't have any serious implications, as I connected your wiki accounts with the Subversion repository (currently using my account for everything, or otherwise I'd have to put your unencrypted password into the wiki database).  But any other visitor might see an outdated page when visiting the wiki.  I will investigate this later, but I gave it a lower priority than before.
